thehoard/thehoard.github.io
# WIP arnoldmasselin/the-11-user-should-be-able-to-download-a-printable-pdf-with-its

### Changes
The PR implements the following features : 
-Select size for the mini based on the generic base from D&D
-the Mini can be placed on an A4 Canvas
-the user can choose how many minis will be added to the canvas
-Another image can be uploaded to create a new mini : the new mini can be added to the pre-existing A4 canvas
-new A4 canveses will be created if the current army does not fit one A4 page
-user can download the army in PDF format


### How to test

- run npm
- type "npm run dev"
- go to your local development server 
- you should see the title Minis-printer with a button to choose an image
- once you choose an image, it will be displayed with a crop selector
- Once you complete the crop, a panel and a preview of the mini will appear
- On the control panel you get to choose the size of the mini : (default value is "moyen") when you change the size, the mini preview should change size.
- If you change the crop, the preview should adjust
- you can choose how many minis will be added to the A4 canvas (default value is 1)
- you can donwload the mini with the provided button, the downloaded mini should have the same features (including size) as the previewed one
- you can add 1 or more minis depending on the value of the field "Nombre de Minis à intégrer" by clicking the button "intégrer le Mini à l'armée"
- when you add one or more mini, the A4 canvas should appear with the minis on it : they will be placed in columns to occupy the height of the canvas.
- Repeat the process with another image : the A4 canvas should keep the previous mini
- repeat the process until the minis does not fit a single A4 page : a new canvas should appear, the previous one should stay
- once the army is complete, vlick the "télécharger l'armée" button : it should download the canvases on a single PDF file

### Minor bugs

-when you crop an image for the first time : the control panel appears but sometimes the crop selector will appear out of place : this doesn't change the actual crop.
-Even though the default size is "moyen" the radio button will not appear checked. (use of the "checked" attribute will make it appear always checked no matter what the user selected)
-when the user already made a mini and want to upload a second image, the preview of the previous mini will stay until the crop is completed on the second image

### Possible evolutions
-Placement algorythm could be optimized  to maximise space use
-a cutting line could be added between every image
